include/firefly/memory-manager/page.hpp

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <cstdlib/cassert.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <atomic>
+#include <utility>
+
+#include "firefly/compiler/clang++.hpp"
+#include "firefly/memory-manager/mm.hpp"
+#include "firefly/stivale2.hpp"
+#include "libk++/bits.h"
+
+static constexpr auto GLOB_PAGE_ARRAY = HIGH_VMA + MiB(512);
+
+enum class RawPageFlags : int {
+    None = 0,
+    Unusable = 1,
+    Slab = 2
+};
+
+struct RawPage {
+    RawPageFlags flags;
+    int order;
+    int buddy_index;
+    std::atomic_int refcount;
+
+    void operator=(const RawPage &other) {
+        flags = other.flags;
+        order = other.order;
+        buddy_index = other.buddy_index;
+        refcount.store(other.refcount, std::memory_order_seq_cst);
+    }
+
+    bool is_buddy_page(int min_order) const {
+        return order >= min_order;
+    }
+    void reset(bool reset_refcount = true) {
+        flags = RawPageFlags::None;
+        order = 0;
+        if (likely(reset_refcount))
+            refcount = 0;
+    }
+} PACKED;
```

Comment:
what's this PACKED?

Reply:
`__attribute__((packed))` in `include/firefly/compiler/clang++.hpp`
It "packs" the struct so that it is as small as possible.

I figured this would be a good idea seeing as we have one `RawPage` for every allocatable physical page

Reply:
Cool, perfect, kinda assumed that.